### PR TITLE
Add eina_thread_id to get Eina_ThreadId from Eina_Thread

### PR DESCRIPTION
**Pull request by cauebs**
The assumption that `Eina_Thread` is a `pthread_t` was already present upstream, so now `Eina_ThreadId` is defined to be just `Eina_Thread`. The assumption that `pthread_t` is an integral type is, however, **wrong according to the standards**, but that might not cause problems on our supported platforms. If we at some point aim to be posix-compliant, we'll have to revisit eina_threads with doubled attention.

(description pulled from a comment below)

Depends on #157

### CODE REVIEW

File: src/tests/eina/eina_suite.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ static const Efl_Test_Case etc[] = {
    { "Module", eina_test_module },
    { "Convert", eina_test_convert },
    { "Rbtree", eina_test_rbtree },
-   { "File", eina_test_file },
+   // { "File", eina_test_file },
```
Comment (Coquinho) [inline src/tests/eina/eina_suite.c:56]: Oh, I think this was for testing, right?

Comment (cauebs) [reply src/tests/eina/eina_suite.c:56]: Oops, this shouldn't have gone through! I'll redo the commit and force push before merging

Comment (Coquinho) [reply src/tests/eina/eina_suite.c:56]: we can just add a new commit, and we can squash it in the end. Do as you prefer.

Comment (cauebs) [reply src/tests/eina/eina_suite.c:56]: Fair enough. Will do.

**Comment by walac** [comment]
The patch, the way it is, makes following comparison be false:
```c
eina_thread_id(eina_thread_self()) == eina_thread_self_id()
````

**Comment by cauebs** [comment]
@walac I addressed that inconsistency by following you suggestion and changing `eina_thread_self_id` to use `pthread_self` on all posix platforms (see last commit).

The assumption that `Eina_Thread` is a `pthread_t` was already present upstream, so now `Eina_ThreadId` is defined to be just `Eina_Thread`. The assumption that `pthread_t` is an integral type is, however, *wrong according to the standards*, but that might not cause problems on our supported platforms. If we at some point aim to be posix-compliant, we'll have to revisit eina_threads with doubled attention.

**Comment by Coquinho** [comment]
Hey, @cauebs this PR has conflicts.

**Comment by Coquinho** [comment]
Wosh, now it is failing.

**Comment by walac** [comment]
This PR should address the change in `eina_self_thread_id` as we discussed in the slack channel.

File: po/POTFILES.in [hunk 2]
```diff
@@ -415,6 +415,7 @@ src/tests/eina/eina_test_quaternion.c
 src/tests/eina/eina_test_strbuf.c
 src/tests/eina/eina_test_lalloc.c
 src/tests/eina/eina_test_bezier.c
+src/tests/eina/eina_test_thread.c
```
Comment (Coquinho) [inline po/POTFILES.in:418]: hey @JPTIZ you know if we have to add things to `po`?

Comment (Coquinho) [reply po/POTFILES.in:418]: @cauebs why you added the test case to `po`?

File: src/tests/eina/eina_test_thread.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2013 ProFUSION embedded systems
```
Comment (Coquinho) [inline src/tests/eina/eina_test_thread.c:2]: @cauebs you can use your name here

File: po/POTFILES.in (hunk 2 above)
Comment (cauebs) [reply po/POTFILES.in:418]: To be quite honest, I just mirrored the other test cases.

File: src/tests/eina/eina_test_thread.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <Eina.h>
+
+#include "eina_suite.h"
+
+EFL_START_TEST(eina_thread_test_self_id)
+{
+    ck_assert(eina_thread_self_id() == eina_thread_id(eina_thread_self()));
```
Comment (JPTIZ) [inline src/tests/eina/eina_test_thread.c:25]: I think you can change this to this:

```suggestion
    ck_assert_int_eq(eina_thread_self_id(), eina_thread_id(eina_thread_self()));
```

(See [libcheck convenience assert functions](https://libcheck.github.io/check/doc/check_html/check_4.html#Convenience-Test-Functions))

Comment (cauebs) [reply src/tests/eina/eina_test_thread.c]: Done, thanks!

File: po/POTFILES.in (hunk 2 above)
Comment (JPTIZ) [reply po/POTFILES.in:418]: > hey @JPTIZ you know if we have to add things to `po`?

Well, we never added a new test file before.

**Comment by Coquinho** [comment]
@cauebs it seems that it got conflicts again

**Comment by joaoantoniocardoso** [comment]
Note that #157 is now approved and if it is merged before this one, we should use `EINA_API` in both source and header files.

**Comment by cauebs** [comment]
> Note that #157 is now approved and if that is merged before this one, we should use `EINA_API` in both source and header files.

Ok, I rebased it with `devs/felipealmeida/native-windows/eina-api`. This PR now depends on #157

**Comment by joaoantoniocardoso** [comment]
Great, to prevent this one to be merged before the other and **also to be easier to receive review**, you can change the base of this PR to the other. It will automatically change its base to native-windows as soon as the other is merged.

**Comment by cauebs** [comment]
TIL. Will do. Thanks, @joaoantoniocardoso

**Pull request merged**